dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -815,3 +876,129 @@ CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtensi
     THROW_IF_WIN32_ERROR(UuidFromStringW(textRpcString, &clsid));
     return clsid;
 }
+
+bool IsOptionEnabled(PCWSTR name)
+{
+    WCHAR value[1 + 1]{};
+    if (GetEnvironmentVariableW(name, value, ARRAYSIZE(value)) == 1)
+    {
+        if (*value == L'0')
+        {
+            return false;
+        }
+        else if (*value == L'1')
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+HRESULT MddBootstrapInitialize_Log(
+    HRESULT hrInitialize,
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion) noexcept try
+{
+    HANDLE hEventLog{ RegisterEventSourceW(nullptr, L"Windows App Runtime") };
+    RETURN_LAST_ERROR_IF_NULL(hEventLog);
+
+    const DWORD c_eventId{ static_cast<DWORD>(hrInitialize) };
+    PCWSTR message1{ L"Windows App Runtime" };
+    WCHAR message2[1024]{};
+    PCWSTR message2Format{ L"ERROR 0x%08X: Bootstrapper initialization failed while looking for version %hu.%hu%s (MSIX packages version >= %hu.%hu.%hu.%hu)" };
+    const UINT16 majorVersion{ HIWORD(majorMinorVersion) };
+    const UINT16 minorVersion{ LOWORD(majorMinorVersion) };
+    WCHAR formattedVersionTag[64]{};
+    if (versionTag && (versionTag[0] != L'\0'))
+    {
+        FAIL_FAST_IF_FAILED(StringCchPrintfW(formattedVersionTag, ARRAYSIZE(formattedVersionTag), L"-%s", versionTag));
+    }
+    FAIL_FAST_IF_FAILED(StringCchPrintfW(message2, ARRAYSIZE(message2), message2Format,
+                                         hrInitialize, majorVersion, minorVersion, formattedVersionTag,
+                                         minVersion.Major, minVersion.Minor, minVersion.Build, minVersion.Revision));
+    PCWSTR strings[2]{ message1, message2 };
+    LOG_IF_WIN32_BOOL_FALSE(ReportEventW(hEventLog, EVENTLOG_ERROR_TYPE, 0, c_eventId, nullptr, ARRAYSIZE(strings), 0, strings, nullptr));
+
+    DeregisterEventSource(hEventLog);
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+HRESULT MddBootstrapInitialize_ShowUI(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion)
+{
+    //TODO: Show this if ERROR_NO_MATCH vs simpler MessageBox("ERROR 0xN in Bootstrapper initialize", MB_OK) ?
```

Comment:
cruft comment?

Reply:
This is actually intentional to draw comment about how we should treat NoMatch vs other errors. See threat above regarding line 118